Improve resolving runtime pack in WasmApp targets

akoeplinger opened the pull request:
Use similar logic to `eng/targetingpacks.targets` in `WasmApp.InTree.targets` and `WasmApp.LocalBuild.targets`.

Also set `UseMonoRuntime=true` to make sure we get the Mono-based runtime pack.

ViktorHofer reviewed (approved):
The targetingpack.targets changes LGTM. Seems like there are some other build failures though.

akoeplinger committed:
Improve resolving runtime pack in WasmApp targets

Import targetingpacks.targets in WasmApp.InTree.targets and use similar logic in WasmApp.LocalBuild.targets.

akoeplinger commented:
Had to take out the `EnableRuntimePackDownload=false` change since it caused `ResolvedRuntimePack` to not be populated so overriding it didn't work anymore.

akoeplinger committed:
Fix runtime Helix tests
